web/src/app/modules/navigation/modules/project-explorer/components/project-explorer.component.html

```diff
@@ -1,48 +1,52 @@
-<div class="input-group">
-    <input #searchBox class="form-control" type="search" (keyup)="search(searchBox.value)" />
-    <div class="input-group-append" title="{{'search' | translate}}">
-        <span class="input-group-text">
-            <i class="fa fa-search"></i>
-        </span>
-    </div>
-</div>
-<ul *ngIf="!searchBox.value">
-    <ngb-tabset>
-        <ngb-tab title="{{'project' | translate}}">
-            <ng-template ngbTabContent>
-                <element-tree *ngFor="let element of rootElements;" [baseUrl]="element.url" [currentElement]="currentElement" [withExpand]="true"></element-tree> 
-                <li *ngIf="canLoadMoreProjectFolders">
-                    <button (click)="loadMoreProjectFolders()" class="btn btn btn-outline-primary btn-sm">
-                        <i class="fa fa-ellipsis-v"></i>&nbsp;{{'loadMore' | translate}}
-                    </button>
-                </li>       
-            </ng-template>
-        </ngb-tab>
-        <ngb-tab title="{{'library' | translate}}">
-            <ng-template ngbTabContent>
-                <element-tree *ngFor="let element of rootLibraries;" [baseUrl]="element.url" [currentElement]="currentElement" [withExpand]="true" [library]="true"></element-tree>        
-                <li *ngIf="canLoadMoreLibraryFolders">
-                    <button (click)="loadMoreLibraryFolders()" class="btn btn btn-outline-primary btn-sm">
-                        <i class="fa fa-ellipsis-v"></i>&nbsp;{{'loadMore' | translate}}
-                    </button>
-                </li>
-            </ng-template>
-        </ngb-tab>
-    </ngb-tabset>
-</ul>
-<table class="table" *ngIf="searchBox.value">
-    <thead>
-        <tr>
-            <th scope="col">{{'searchResults' | translate}}</th>
-        </tr>
-    </thead>
-    <tbody>
-        <tr *ngFor="let element of searchResults; let i = index;">
-            <td>
-                <a [navigationTarget]="element">
-                    <icon-selector [model]="element" expanded="false"></icon-selector>{{element['extId'] ? element['extId'] + ': ' : ''}}{{element.name}}
-                </a>
-            </td>
-        </tr>
-    </tbody>
-</table>
+<div class="input-group">
+    <input #searchBox class="form-control" type="search" (keyup)="search(searchBox.value)" />
+    <button *ngIf="searchBox.value"
+        (click)="searchBox.value = null" title="{{'delete' | translate}}"

```

Comment:
please make this a code-behind function (e.g. "clear search")

Reply:
Not quite sure what I did there now, but it works. Please tell me, if you know a nicer way to do this.